### PR TITLE
FOGL-1773 sense HAT plugin code added with debian package structure

### PR DESCRIPTION
**Pull request by ashish-jabble**
**Features**

The Sense HAT features an 8x8 RGB LED matrix, a mini joystick and the following sensors:

- [x] Gyroscope
- [x] Accelerometer
- [x] Magnetometer
- [x] Temperature
- [x] Humidity
- [x] Barometric pressure

Do we really need to implement 8x8 RGB LED and joystick implementation?

- [x] joystick

**NOTE** - While stopping the server, you may see and this will be handled separately
```
ERROR: server: foglamp.services.south.server: Failed to poll for plugin sense
Sep  1 11:39:27 raspberrypi FogLAMP[7528] ERROR: server: foglamp.services.south.server: Max retries exhausted in starting South plugin: sense#012Traceback (most recent call last):#012  File "/usr/lib/python3.5/asyncio/tasks.py", line 304, in _wakeup#012    future.result()#012  File "/usr/lib/python3.5/asyncio/futures.py", line 285, in result#012    raise CancelledError#012concurrent.futures._base.CancelledError
```

### CODE REVIEW

File: packages/Debian/armhf/DEBIAN/preinst [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/bin/sh
```
Comment (praveen-garg) [inline packages/Debian/armhf/DEBIAN/preinst:1]: remove this

File: python/foglamp/plugins/south/sensehat/sensehat.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    },
+    'magnetometerSensorName': {
+        'description': 'Asset name of magnetometer sensor',
+        'type': 'string',
+        'default': 'magnetometer',
+        'order': '14'
+    }
+}
+
+_LOGGER = logger.setup(__name__, level=20)
```
Comment (praveen-garg) [inline python/foglamp/plugins/south/sensehat/sensehat.py]: always better to import logging and set level=logging.INFO

File: python/foglamp/plugins/south/sensehat/sensehat.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+    Args:
+        handle: handle returned by the plugin initialisation call
+    Returns:
+        returns a sensor reading in a JSON document, as a Python dict, if it is available
+        None - If no reading is available
+    Raises:
+        DataRetrievalError
+    """
+    sense = SenseHat()
```
Comment (praveen-garg) [inline python/foglamp/plugins/south/sensehat/sensehat.py]: could you please make it `if not sense: sense = SenseHat()`

File: python/foglamp/plugins/south/sensehat/sensehat.py [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+            accelerometer = sense.get_accelerometer_raw()
+            data.append({
+                'asset': '{}{}'.format(asset_prefix, handle['accelerometerSensorName']['value']),
+                'timestamp': time_stamp,
+                'key': str(uuid.uuid4()),
+                'readings': accelerometer
+            })
+
+    except (Exception, RuntimeError) as ex:
+        _LOGGER.exception("Sense HAT exception: {}".format(str(ex)))
```
Comment (praveen-garg) [inline python/foglamp/plugins/south/sensehat/sensehat.py]: I suggest to handle `RuntimeError as e` in other block, you don't need str(ex). always use %s for logger

File: python/foglamp/plugins/south/sensehat/sensehat.py [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+        raise exceptions.DataRetrievalError(ex)
+
+    return data
+
+
+def plugin_reconfigure(handle, new_config):
+    """  Reconfigures the plugin
+
+    it should be called when the configuration of the plugin is changed during the operation of the South device service;
```
Comment (praveen-garg) [inline python/foglamp/plugins/south/sensehat/sensehat.py]: `device` -> `plugin`

File: python/foglamp/plugins/south/sensehat/sensehat.py [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+
+    Args:
+        handle: handle returned by the plugin initialisation call
+    Returns:
+
+    """
+    _LOGGER.info('Sense HAT poll plugin shut down.')
+
+
+def _str_to_bool(s):
```
Comment (praveen-garg) [inline python/foglamp/plugins/south/sensehat/sensehat.py]: move it as internal def of `plugin_poll`

**Review by praveen-garg — CHANGES_REQUESTED** [review]

**Comment by billehunt** [comment]
re: Do we really need to implement 8x8 RGB LED and joystick implementation?

For now, no. However, @MarkRiddoch, we should think about how to fit control into our architecture.

**Comment by billehunt** [comment]
Actually, please implement joystick as the API returns: {"direction": "up", "action": "pressed"}.

Functionally, this looks great. Please resolve the feedback called out in this PR and commit it to the develop branch ASAP.

**Comment by ashish-jabble** [comment]
@billehunt @praveen-garg feedback changes incorporated and joystick feature implemented.

**Review by praveen-garg — APPROVED** [review]

**Comment by praveen-garg** [comment]
Merging ASAP. 
May be we need to  handle removal of `preinstl` script and debian package test later!

**Pull request merged**